File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    
    # for 문 전체 순회
    # 인덱스 1부터, 전 원소와 같으면 스킵
    # 다르면 answer에 append
    answer.append(arr[0])
    for idx, i in enumerate(arr, start = 1):
        if (i != answer[-1]):
            answer.append(i)


    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> list = new ArrayList<>();
        
        // a = -1로 설정
        // for문 돌면서 다음 값이 a와 다르면 answer에 추가
        // 같으면 그냥 인덱스 +1
        int a = -1;
        for (int i = 0; i<arr.length ;i++)
            if (arr[i] != a){
                list.add(arr[i]);
                a = arr[i];
            }
                
        int[] answer = new int[list.size()];
        for (int i = 0; i<list.size(); i++){
            answer[i] = list.get(i);
        }
        
        
        return answer;
    }
}